### PR TITLE
[WIP] Add support for configuring Google OAuth logins

### PR DESCRIPTION
**Pull request by alexpdp7**
Known issues:

* Upon registration/login you are redirected to an ugly error message.
  If you change the URL you can access the application and things are
  working, though

* User cannot access the user list!

* No documentation for the new environment variables

Signed-off-by: Alex Corcoles <alex@pdp7.net>

---

* [ ] I have updated the CHANGELOG file according to the conventions in [keepachangelog.com](https://keepachangelog.com)
* [ ] This PR contains changes that do not require a mention in the CHANGELOG file

### CODE REVIEW

File: superset/contrib/docker/superset_config.py [hunk 1]
```diff
@@ -152,3 +152,26 @@ def mutator(f):
 # This parameter documented as deprecated but frontend still uses it
 # Should match `--timeout` value of gunicorn
 SUPERSET_WEBSERVER_TIMEOUT = 300
+
+if get_env_variable('OAUTH_PROVIDER', False) == 'google':
```
Comment (smacker) [inline superset/contrib/docker/superset_config.py]: maybe we should exit with an error if `OAUTH_PROVIDER` value is unknown.

File: superset/contrib/docker/superset_config.py [hunk 2]
```diff
@@ -152,3 +152,26 @@ def mutator(f):
...
+    from flask_appbuilder.security.manager import AUTH_OAUTH
+
+    AUTH_TYPE = AUTH_OAUTH
+
+    OAUTH_PROVIDERS = [
+        {'name':'google', 'icon':'fa-google', 'token_key':'access_token',
+            'remote_app': {
+                'consumer_key': get_env_variable('OAUTH_CONSUMER_KEY'),
+                'consumer_secret': get_env_variable('OAUTH_CONSUMER_SECRET'),
+                'base_url':'https://www.googleapis.com/oauth2/v2/',
```
Comment (smacker) [inline superset/contrib/docker/superset_config.py]: please format the code

**Comment by se7entyse7en** [comment]
From [private] [slack](https://src-d.slack.com/archives/C7TB5NEDN/p1570023247201100?thread_ts=1570018104.198500&cid=C7TB5NEDN):

> IMO it makes sense to open a PR on sourced-ui pointing to your PR :thumbsup::skin-tone-4: where we could eventually add some refinements

Actually I meant issue 😅 but I wrote PR. I think that we could use this as a base and take ownership. But there's no urgency for this to be included now.

File: superset/contrib/docker/superset_config.py (hunk 1 above)
Comment (alexpdp7) [reply superset/contrib/docker/superset_config.py]: I've done a draft implementation of that, although I don't love it

File: superset/contrib/docker/superset_config.py (hunk 2 above)
Comment (alexpdp7) [reply superset/contrib/docker/superset_config.py]: It passes flake8 now (I didn't find a specific flake8 config in the repo)

**Comment by alexpdp7** [comment]
> Actually I meant issue sweat_smile but I wrote PR. I think that we could use this as a base and take ownership. But there's no urgency for this to be included now.

Don't worry, it took minutes to do it. It still lacks documentation but at least you have something to start with, I guess

File: superset/contrib/docker/superset_config.py (hunk 2 above)
Comment (smacker) [reply superset/contrib/docker/superset_config.py]: https://github.com/src-d/sourced-ui/pull/292/checks?check_run_id=244877259
sourced-ui uses flake8, pylint and black.

Comment (alexpdp7) [reply superset/contrib/docker/superset_config.py]: flake8 and pylint look fine for this change (although I'm not 100% familiar with pylint).

This file doesn't follow black conventions, so I'd leave it as-is, but if you want, I can introduce a commit before applying black to the existing file, then add my change with black conventions.

**Comment by smacker** [comment]
another option, I can just close it, copy-paste the code into my PR and take all the praise for the implementation of the feature 😝

**Comment by alexpdp7** [comment]
:D

I really don't mind. I've had success with using git format-patch / am to get out of these situations. But I'm not proud of that- I feel like I'm doing a dumb thing instead of the "proper" solution.

**Comment by se7entyse7en** [comment]
hahaha

I'd vote for this BTW:

> another option, I can just close it, copy-paste the code into my PR and take all the praise for the implementation of the feature 😝

😂 

I prefer to avoid having non-properly ready things.

**Pull request closed without merging**